### ISSUE TITLE
Changed behavior with wasabi gem version 3.6.1

### ISSUE THREAD

**Issue by cotelha** [savonrb/savon#948]
# Bug report

**Current behavior:**
I get an error when Savon(2.12.1) using the gem wasabi version 3.6.1, but when I use wasabi version 3.5.0 it works.
Wasabi 3.6.1 changed the behavior that reading the method https://github.com/savonrb/savon/blob/v2.12.1/lib/savon/builder.rb#L184 (`@wsdl.soap_input`) now always returns a hash in my case.

**Steps to reproduce current behavior:**
I used example: https://gist.github.com/cotelha/7a49e8561d8a49d1a0eabbf8216b3b73
With Wasabi 3.5.0
```
2.5.5 :004 > document = Wasabi.document File.read("...IdmDados.xml")
2.5.5 :004 > document.soap_input(:requisicao)
 => "requisicao" 
```

With Wasabi 3.6.1
```
2.5.5 :004 > document = Wasabi.document File.read("...IdmDados.xml")
2.5.5 :004 > document.soap_input(:requisicao)
 => {"requisicao"=>{"XML"=>["xs", "string"]}} 
```

With Savon(2.12.1) and Wasabi(3.6.1)
```
2.5.5 :004 > cli = Savon.client(wsdl: "https://www.foo.com.br/wbsStoreage/bar.exe/wsdl/IdmDados").
call(:requisicao, message: '<XML xsi:type="xsd:string">&lt;?xml version="1.0" encoding="iso-8859-1"?&gt;&lt;requisicao in_schema="F"&gt;&lt;loginws cd_loginws="foo" cd_senhaws="bar"/&gt;&lt;pessoa acao="con" nr_cpfcnpj="000000000"/&gt;&lt;/requisicao&gt;</XML>')
Traceback (most recent call last):
        1: from (irb):48
TypeError (no implicit conversion of Symbol into Integer)
```

**Expected behavior:**
I expected that the new version of the wasabi gem does not change the behavior the Savon.

**System information:**

* ruby version: 2.5.5
* savon version: 2.12.1

**Comment by joshRpowell** [comment]
reference #935

**Comment by diesl** [comment]
Not stale

**Comment by olleolleolle** [comment]
For those investigating, perhaps the changes in behavior made to https://github.com/savonrb/wasabi/pull/103#issuecomment-953024540 can have impact here, too? Please report back if the current latest code in the git repo works for your use-case.

**Comment by olleolleolle** [comment]
📓 If you want to create a runnable example: https://bundler.io/guides/bundler_in_a_single_file_ruby_script.html Bundler has an inline mode, which is perfect for that kind of thing.

**Comment by olleolleolle** [comment]
Perhaps the v3.7.0, released today, is worth trying out, too?

**Comment by pcai** [comment]
I believe this was fixed in https://github.com/savonrb/wasabi/pull/103 which was released in wasabi 3.7.0. Can you please reopen with repro information if this is still not working?
